runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/DokkaCollectorTask.kt

```diff
@@ -13,42 +11,44 @@ open class DokkaCollectorTask : DefaultTask() {
     var modules: List<String> = emptyList()
 
     @Input
-    var outputDirectory: String = ""
+    var outputDirectory: String = defaultDokkaOutputDirectory().absolutePath
 
     private lateinit var configuration: GradleDokkaConfigurationImpl
 
+    @Input
+    val dokkaTaskNames: MutableSet<String> = mutableSetOf()
+
     @TaskAction
     fun collect() {
-        val sourceSets = getProjects(project).filter { it.name in modules }.flatMap {
-            val tasks = try {
-                it.tasks.withType(DokkaTask::class.java)
-            } catch (e: UnknownTaskException) {
-                throw IllegalStateException("No dokka task declared in module ${it.name}")
-            }
-            tasks.map { it.getConfigurationOrNull() }
-        }.filterNotNull()
+        val configurations = project.subprojects
+            .filter { subProject -> subProject.name in modules }
+            .flatMap { subProject -> dokkaTaskNames.mapNotNull(subProject.tasks::findByName) }
+            .filterIsInstance<DokkaTask>()
+            .mapNotNull { dokkaTask -> dokkaTask.getConfigurationOrNull() }
+
 
         val initial = GradleDokkaConfigurationImpl().apply {
             outputDir = outputDirectory
-            cacheRoot = sourceSets.first().cacheRoot
-            format = sourceSets.first().format
+            cacheRoot = configurations.first().cacheRoot
         }
 
-        configuration = sourceSets.fold(initial) { acc, it: GradleDokkaConfigurationImpl ->
-            if(acc.format != it.format || acc.cacheRoot != it.cacheRoot)
-                throw IllegalStateException("Dokka task configurations differ on core arguments (format, cacheRoot)")
+        // TODO this certainly not the ideal solution
+        configuration = configurations.fold(initial) { acc, it: GradleDokkaConfigurationImpl ->
+            if (acc.cacheRoot != it.cacheRoot)
+                throw IllegalStateException("Dokka task configurations differ on core argument cacheRoot")
             acc.sourceSets = acc.sourceSets + it.sourceSets
             acc.pluginsClasspath = (acc.pluginsClasspath + it.pluginsClasspath).distinct()
             acc
         }
-        project.tasks.withType(DokkaTask::class.java).configureEach { it.config = configuration }
+        project.tasks.withType(DokkaTask::class.java).configureEach { it.enforcedConfiguration = configuration }
     }
 
     init {
-        finalizedBy(project.tasks.getByName(DOKKA_TASK_NAME))
+        // TODO: This this certainly not the ideal solution
+        dokkaTaskNames.forEach { dokkaTaskName ->
+            finalizedBy(dokkaTaskName)
+        }
```

Comment:
Can we get those tasks and `dependOn` them instead? (I know that this was introduced in another commit, just wondering if we can fix it RN)

Reply:
I think the DokkaCollectorTask deserves to be entirely re-designed as well. Changing the configuration of another task during execution and then running those seems really wrong to me. 

I am to afraid to change this RN, but I want to tackle this as well in a later PR!